re-organzied port-agent and added argocd installation method (#762)
port-labs/port-docs: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md

---
sidebar_position: 2
---

# ArgoCD

> Let's dive in to a walkthrough on how to install the Port execution agent in your Kubernetes cluster.

Reviewer: You used the same sentence here and in the Helm page. Perhaps let's change it to something like:
This page will walk you through the installation of the Port execution agent in your Kubernetes cluster using ArgoCD.

And let's use the same format in the Helm page.

:::info

> You can observe the helm chart with the full installation [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).

Reviewer 2: I altered the sentence slightly in the comment on the helm page, let's apply it here too

:::

:::note prerequisites

> - [ArgoCD](https://argoproj.github.io/cd/) must be installed in your Kubernetes cluster. Please refer to
>   ArgoCD's [documentation](https://argo-cd.readthedocs.io/en/stable/) for further details on the installation;

Reviewer: ```suggestion
- [ArgoCD](https://argoproj.github.io/cd/) must be installed in your Kubernetes cluster. Please refer to the [ArgoCD documentation](https://argo-cd.readthedocs.io/en/stable/) for further details about the installation.
```

:::

## Installation

Install the `my-port-agent` ArgoCD Application by using the following yaml:
:::note
Remember to replace the placeholders for `YOUR_ORG_ID`, `YOUR_KAFKA_CONSUMER_GROUP`, `YOUR_PORT_CLIENT_ID` and `YOUR_PORT_CLIENT_SECRET`.
:::

<details>
  <summary>ArgoCD Application</summary>

```yaml showLineNumbers
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: my-port-agent
spec:
  destination:
    name: ''
    namespace: my-port-agent
    server: 'https://kubernetes.default.svc'
  source:
    path: ''
    repoURL: 'https://port-labs.github.io/helm-charts'
    targetRevision: 0.7.2
    chart: port-agent
    helm:
      parameters:
        - name: env.normal.KAFKA_CONSUMER_GROUP_ID
          value: YOUR_KAFKA_CONSUMER_GROUP
        - name: env.normal.PORT_ORG_ID
          value: YOUR_ORG_ID
        - name: env.secret.PORT_CLIENT_ID
          value: YOUR_PORT_CLIENT_ID
        - name: env.secret.PORT_CLIENT_SECRET
          value: YOUR_PORT_CLIENT_SECRET
  sources: []
  project: default
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
    syncOptions:
      - CreateNamespace=true
```

</details>
<br/>

## Next Steps

> - [Usage](/create-self-service-experiences/self-service-actions-deep-dive/self-service-actions-deep-dive.md) guide Set up a webhook.

Reviewer: This is unclear - what is the step? The link just takes me to the deep dive page. Also, the sentence is incorrect - "Usage guide set up a webhook". Please revise:)

```

3. Commit the changes to your git repository.

4. Install the `my-port-agent` ArgoCD Application by using the following yaml:
:::note
Remember to replace the placeholders for `YOUR_ORG_ID`, `YOUR_KAFKA_CONSUMER_GROUP`, `YOUR_PORT_CLIENT_ID` `YOUR_PORT_CLIENT_SECRET` and `YOUR_GIT_REPO_URL`.
:::

<details>
  <summary>ArgoCD Application</summary>

```yaml showLineNumbers
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: my-port-agent
  namespace: argocd
spec:
  destination:
    namespace: my-port-agent
    server: https://kubernetes.default.svc
  project: default
  source:
    helm:
      parameters:
        - name: env.normal.KAFKA_CONSUMER_GROUP_ID
          value: YOUR_KAFKA_CONSUMER_GROUP
        - name: env.normal.PORT_ORG_ID
          value: YOUR_ORG_ID
        - name: env.secret.PORT_CLIENT_ID
          value: YOUR_PORT_CLIENT_ID
        - name: env.secret.PORT_CLIENT_SECRET
          value: YOUR_PORT_CLIENT_SECRET
    path: argocd/port-agent
    repoURL: YOUR_GIT_REPO_URL
    targetRevision: main
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
    syncOptions:
    - CreateNamespace=true
```

</details>
<br/>

## Next Steps

> - Refer to the [Usage](/create-self-service-experiences/setup-backend/webhook/port-execution-agent/usage.md) guide Set up a webhook.

Reviewer: ```suggestion
- Refer to the [usage guide](/create-self-service-experiences/setup-backend/webhook/port-execution-agent/usage.md) to set up a webhook.
```
